components/brave_wallet/browser/bitcoin/bitcoin_wallet_service.cc

```diff
@@ -691,6 +703,156 @@ void DiscoverNextUnusedAddressTask::OnGetAddressStats(
   WorkOnTask();
 }
 
+class DiscoverAccountTask : public base::RefCounted<DiscoverAccountTask> {
```

Comment:
why are all of these RefCounted? Refcounting is discouraged and some of these are RefCountedThreadSafe (like GetBalanceTask above), but they do not appear to actually be thread safe which is highly concerning. 

https://www.chromium.org/developers/coding-style/important-abstractions-and-data-structures/#scoped_refptrt-baserefcounted-baserefcountedthreadsafe

```
Reference counting is occasionally useful but is more often a sign that someone isn't thinking carefully about ownership. Use it when ownership is truly shared (for example, multiple tabs sharing the same renderer process), not for when lifetime management is difficult to reason about.
```